webui/react/src/pages/WorkspaceList.tsx

```diff
@@ -1,21 +1,336 @@
+import SelectFilter from 'components/SelectFilter';
+import Spinner from 'components/Spinner';
+import { GenericRenderer, getFullPaginationConfig, userRenderer } from 'components/Table';
+import { useStore } from 'contexts/Store';
 import useModalWorkspaceCreate from 'hooks/useModal/Workspace/useModalWorkspaceCreate';
+import usePolling from 'hooks/usePolling';
+import useResize from 'hooks/useResize';
+import useSettings, { UpdateSettings } from 'hooks/useSettings';
+import { paths } from 'routes/utils';
+import { getWorkspaces } from 'services/api';
+import { V1GetWorkspacesRequestSortBy } from 'services/api-ts-sdk';
+import { validateDetApiEnum } from 'services/utils';
+import { ShirtSize } from 'themes';
+import { Workspace } from 'types';
+import { isEqual } from 'utils/data';
+
+import css from './WorkspaceList.module.scss';
+import settingsConfig, { DEFAULT_COLUMN_WIDTHS,
+  WorkspaceColumnName, WorkspaceListSettings } from './WorkspaceList.settings';
+import WorkspaceActionDropdown from './WorkspaceList/WorkspaceActionDropdown';
+import WorkspaceCard from './WorkspaceList/WorkspaceCard';
+
+const { Option } = Select;
+
+enum WorkspaceFilters {
+  All = 'ALL_WORKSPACES',
+  Mine = 'MY_WORKSPACES',
+  Others = 'OTHERS_WORKSPACES'
+}
+
+/*
+ * This indicates that the cell contents are rightClickable
+ * and we should disable custom context menu on cell context hover
+ */
+const onRightClickableCell = () =>
+  ({ isCellRightClickable: true } as React.HTMLAttributes<HTMLElement>);
 
 const WorkspaceList: React.FC = () => {
+  const { users, auth: { user } } = useStore();
   const { modalOpen } = useModalWorkspaceCreate({});
+  const [ workspaces, setWorkspaces ] = useState<Workspace[]>([]);
+  const [ workspaceFilter, setWorkspaceFilter ] = useState<WorkspaceFilters>(WorkspaceFilters.All);
+  const [ total, setTotal ] = useState(0);
+  const [ pageError, setPageError ] = useState<Error>();
+  const [ isLoading, setIsLoading ] = useState(true);
+  const pageRef = useRef<HTMLElement>(null);
+  const [ canceler ] = useState(new AbortController());
+  const size = useResize();
+
+  const {
+    settings,
+    updateSettings,
+  } = useSettings<WorkspaceListSettings>(settingsConfig);
 
   const handleWorkspaceCreateClick = useCallback(() => {
     modalOpen();
   }, [ modalOpen ]);
+
+  const fetchWorkspaces = useCallback(async () => {
+    try {
+      const response = await getWorkspaces({
+        archived: settings.archived ? undefined : false,
+        limit: settings.tableLimit,
+        name: settings.name,
+        offset: settings.tableOffset,
+        orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+        sortBy: validateDetApiEnum(V1GetWorkspacesRequestSortBy, settings.sortKey),
+        users: settings.user,
+      }, { signal: canceler.signal });
+      setTotal(response.pagination.total ?? 0);
+      setWorkspaces(prev => {
+        const withoutDefault = response.workspaces.filter(w => !w.immutable);
```

Comment:
I have Uncategorized missing from the workspaces list right now, I think from this filter here.

Reply:
Yes, this is intentional. The Uncategorized (default) workspace should not show up in the workspaces list. It will have a separate link in the navbar (to be added later).

Reply:
Ah, ok. Thanks